ios/RNSentry.m

```diff
@@ -31,56 +31,70 @@ + (BOOL)requiresMainQueueSetup {
                   resolve:(RCTPromiseResolveBlock)resolve
                   rejecter:(RCTPromiseRejectBlock)reject)
 {
     NSError *error = nil;
-    SentryClient *client = [[SentryClient alloc] initWithDsn:dsnString didFailWithError:&error];
-    client.shouldSendEvent = ^BOOL(SentryEvent * _Nonnull event) {
+
+    SentryBeforeSendEventCallback beforeSend = ^SentryEvent*(SentryEvent *event) {
         // We don't want to send an event after startup that came from a Unhandled JS Exception of react native
         // Because we sent it already before the app crashed.
         if (nil != event.exceptions.firstObject.type &&
             [event.exceptions.firstObject.type rangeOfString:@"Unhandled JS Exception"].location != NSNotFound) {
             NSLog(@"Unhandled JS Exception");
-            return NO;
+            return nil;
         }
-        return YES;
+
+        return event;
     };
-    [SentryClient setSharedClient:client];
-    if ([[options objectForKey:@"enableNativeCrashHandling"] boolValue]) {
-        [SentryClient.sharedClient startCrashHandlerWithError:&error];
-    }
-    if (nil != [options objectForKey:@"environment"]) {
-        SentryClient.sharedClient.environment = [NSString stringWithFormat:@"%@", [options objectForKey:@"environment"]];
-    }
-    if (nil != [options objectForKey:@"release"]) {
-        SentryClient.sharedClient.releaseName = [NSString stringWithFormat:@"%@", [options objectForKey:@"release"]];
-    }
-    if (nil != [options objectForKey:@"dist"]) {
-        SentryClient.sharedClient.dist = [NSString stringWithFormat:@"%@", [options objectForKey:@"dist"]];
-    }
+
+    [options setValue:beforeSend forKey:@"beforeSend"];
+
+    SentryOptions *sentryOptions = [[SentryOptions alloc] initWithDict:options didFailWithError:&error];
     if (error) {
         reject(@"SentryReactNative", error.localizedDescription, error);
         return;
     }
+    [SentrySDK startWithOptionsObject:sentryOptions];
+
     resolve(@YES);
 }
 
 RCT_EXPORT_METHOD(deviceContexts:(RCTPromiseResolveBlock)resolve
                   rejecter:(RCTPromiseRejectBlock)reject)
 {
-    resolve([[[SentryContext alloc] init] serialize]);
+    NSLog(@"Bridge call to: deviceContexts");
+    // Temp work around until sorted out this API in sentry-cocoa.
+    // TODO: If the callback isnt' executed the promise wouldn't be resolved.
+    [SentrySDK configureScope:^(SentryScope * _Nonnull scope) {
+        NSDictionary<NSString *, id> *serializedScope = [scope serialize];
+        // Scope serializes as 'context' instead of 'contexts' as it does for the event.
+        NSDictionary<NSString *, id> *contexts = [serializedScope valueForKey:@"context"];
+#if DEBUG
+        NSData *data = [NSJSONSerialization dataWithJSONObject:contexts options:0 error:nil];
+        NSString *debugContext = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+        NSLog(@"Contexts: %@", debugContext);
+#endif
+        resolve(contexts);
```

Comment:
It's not stupid if it works.
And tbh we probably need to do a few of those "hacks" in order to make it work as needed so I am OK